instrumentation/spring-amqp/src/main/test/brave/spring/amqp/ITSpringAmqpTracing.java

```diff
@@ -0,0 +1,222 @@
+    }
+
+    @Bean
+    public Queue queue() {
+      return new Queue("test-queue");
+    }
+
+    @Bean
+    public Binding binding(Exchange exchange, Queue queue) {
+      return BindingBuilder.bind(queue).to(exchange).with("test.binding").noargs();
+    }
+
+    @Bean
+    public AmqpAdmin amqpAdmin(ConnectionFactory connectionFactory) {
+      return new RabbitAdmin(connectionFactory);
+    }
+  }
+
+  @Configuration
+  public static class TracingProducerConfig {
+    @Bean
+    public Tracing tracing(Reporter<Span> reporter) {
+      return Tracing.newBuilder()
+          .localServiceName("spring-amqp-producer")
+          .sampler(Sampler.ALWAYS_SAMPLE)
+          .spanReporter(reporter)
+          .build();
+    }
+
+    @Bean
+    public TracingMessagePostProcessor tracingMessagePostProcessor(Tracing tracing) {
+      return new TracingMessagePostProcessor(tracing);
+    }
+
+    @Bean
+    public RabbitTemplate rabbitTemplate(ConnectionFactory connectionFactory, TracingMessagePostProcessor tracingMessagePostProcessor) {
+      RabbitTemplate rabbitTemplate = new RabbitTemplate(connectionFactory);
+      rabbitTemplate.setBeforePublishPostProcessors(tracingMessagePostProcessor);
+      rabbitTemplate.setExchange("test-exchange");
+      return rabbitTemplate;
+    }
+
+    @Bean
+    public Reporter<Span> reporter(List<Span> spans) {
+      return spans::add;
+    }
+
+    @Bean
+    public List<Span> producerSpans() {
+      return new ArrayList<>();
+    }
+
+    @Bean
+    public TracingRabbitProducer tracingRabbitProducer(RabbitTemplate rabbitTemplate) {
+      return new TracingRabbitProducer(rabbitTemplate);
+    }
+  }
+
+  @EnableRabbit
+  @Configuration
+  public static class TracingConsumerConfig {
+    @Bean
+    public Tracing tracing(List<Span> spans) {
+      return Tracing.newBuilder()
+          .localServiceName("spring-amqp-consumer")
+          .sampler(Sampler.ALWAYS_SAMPLE)
+          .spanReporter(spans::add)
+          .build();
+    }
+
+    @Bean
+    public List<Span> consumerSpans() {
+      return new ArrayList<>();
+    }
+
+    @Bean
+    public TracingRabbitListenerAdvice tracingRabbitListenerAdvice(Tracing tracing) {
+      return new TracingRabbitListenerAdvice(tracing);
+    }
+
+    @Bean
+    public SimpleRabbitListenerContainerFactory rabbitListenerContainerFactory(
+        ConnectionFactory connectionFactory, Advice tracingRabbitListenerAdvice) {
+      SimpleRabbitListenerContainerFactory factory = new SimpleRabbitListenerContainerFactory();
+      factory.setConnectionFactory(connectionFactory);
+      factory.setAdviceChain(tracingRabbitListenerAdvice);
+      return factory;
+    }
+
+    @Bean
+    public TracingRabbitConsumer tracingRabbitConsumer(CountDownLatch countDownLatch) {
+      return new TracingRabbitConsumer(countDownLatch);
+    }
+
+    @Bean CountDownLatch messageReceivedLatch() {
+      return new CountDownLatch(1);
+    }
+  }
+
+  private static class TracingRabbitProducer {
```

Comment:
nit: in your test class avoid the prefix Tracing unless it is an instrumented type. That's because our normal instrumentation types are prefixed Tracing. I was looking to see if this class was in main. Maybe instead use just "RabbitProducer" or "HelloWorldProducer"